core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -966,20 +974,22 @@ public static void clearBlockCache(List<AbstractIndex> dataBlocks) {
     List<Boolean> isDictionaryDimensions = new ArrayList<Boolean>();
     Set<Integer> processedColumnGroup = new HashSet<Integer>();
     for (CarbonDimension carbonDimension : tableDimensionList) {
-      List<CarbonDimension> childs = carbonDimension.getListOfChildDimensions();
-      //assuming complex dimensions will always be atlast
-      if (null != childs && childs.size() > 0) {
-        break;
-      }
-      if (carbonDimension.isColumnar() && hasEncoding(carbonDimension.getEncoder(),
-          Encoding.DICTIONARY)) {
-        isDictionaryDimensions.add(true);
-      } else if (!carbonDimension.isColumnar()) {
-        if (processedColumnGroup.add(carbonDimension.columnGroupId())) {
+      if (!carbonDimension.isInvisible()) {
```

Comment:
The list which is being created, should have two getters. one to get With invisible and other to get without invisible. Use required function accordingly